File: BabylonProject/Assets/Project/Scripts/MyRotator.cs
using System;
using UnityEditor;
using UnityEngine;

/**
* Editor Script Component
* @class MyRotator
*/
[Babylon(Class="PROJECT.MyRotator"), AddComponentMenu("Scripts/My Project/MyRotator")]
public class MyRotator : EditorScriptComponent
{
    
    [Auto] [Range(0.0f,10.0f)] public float rotateSpeed = 0.25f;

}

File: BabylonProject/Assets/Project/Scripts/MyRotator.ts
module PROJECT {
    /**
    * Babylon Script Component
    * @class MyRotator
    */
    export class MyRotator extends BABYLON.Toolkit.ScriptComponent {
        public rotateSpeed : number = 0.25;
        protected update(): void {
            /* Update render loop function */
            const delta:number = this.getDeltaSeconds();
            const roate:number = this.rotateSpeed * delta;
            this.transform.addRotation(0,roate,0);
        }

    }
}